butterfly-utilities/src/main/java/com/paypal/butterfly/utilities/operations/xml/XmlXPathReplace.java

```diff
@@ -0,0 +1,184 @@
+package com.paypal.butterfly.utilities.operations.xml;
+
+import java.io.File;
+import java.io.IOException;
+
+import javax.xml.parsers.DocumentBuilder;
+import javax.xml.parsers.DocumentBuilderFactory;
+import javax.xml.parsers.ParserConfigurationException;
+import javax.xml.transform.Transformer;
+import javax.xml.transform.TransformerConfigurationException;
+import javax.xml.transform.TransformerException;
+import javax.xml.transform.TransformerFactory;
+import javax.xml.transform.dom.DOMSource;
+import javax.xml.transform.stream.StreamResult;
+import javax.xml.xpath.XPath;
+import javax.xml.xpath.XPathConstants;
+import javax.xml.xpath.XPathExpression;
+import javax.xml.xpath.XPathExpressionException;
+import javax.xml.xpath.XPathFactory;
+
+import org.w3c.dom.Document;
+import org.w3c.dom.Element;
+import org.w3c.dom.Node;
+import org.w3c.dom.NodeList;
+import org.xml.sax.SAXException;
+
+import com.paypal.butterfly.extensions.api.TOExecutionResult;
+import com.paypal.butterfly.extensions.api.TransformationContext;
+import com.paypal.butterfly.extensions.api.TransformationOperation;
+import com.paypal.butterfly.extensions.api.exception.TransformationDefinitionException;
+import com.paypal.butterfly.extensions.api.exception.TransformationOperationException;
+import com.paypal.butterfly.extensions.api.exception.TransformationUtilityException;
+
+/**
+ * Finds a particular xpath in an XML file and can operate two different modes
+ * <br>
+ * If an {@link org.w3c.dom.Element} is provided, the whole node is replaced by the element.
+ * If a string is provided, then the text of the node is replaced by the string.
+ * <br>
+ * If the xpath expression won't compile, an error is returned.
+ * <br>
+ * If the file is not a well formed XML file, an error is returned.
+ *
+ * @author mmcrockett
+ */
+public class XmlXPathReplace extends TransformationOperation<XmlXPathReplace> {
+    private static final String DESCRIPTION = "Replace %s of xpath %s in XML file %s with %s";
+
+    private String xpathExpressionString;
+    private XPathExpression xpathExpression;
+    private Object replacementObject;
```

Comment:
Should not this be `Element` instead of `Object`?

Reply:
Also, it should be called `replacementElement`, according to the constructor.

Reply:
Oh, I see, you are keeping only one attribute regardless of the replacement type. That is ok then.